### ISSUE TITLE
Database location issue

### ISSUE THREAD

**Issue by scgough** [andpor/react-native-sqlite-storage#49]
Hi there,

Note: Running v2.1.3

I've just been trying to move my database to a non-icloud backed up location in accordance with app submission new rules and also as per the cordova docs here:
```
with the location option set to one the following choices (affects iOS only):
0 (default): Documents - visible to iTunes and backed up by iCloud
1: Library - backed up by iCloud, NOT visible to iTunes
2: Library/LocalDatabase - NOT visible to iTunes and NOT backed up by iCloud (same as using "default")
```

I managed to figure that this is achieved via the `dblocation:'nosync'` setting when opening a database but on the native side this is seemingly ignored and always has the value `docs`

I've had to hard-code `nosync` in to the native side in `SQLite.m`

**Comment by andpor** [comment]
I will look into this.

**Comment by grittathh** [comment]
I have a sort of related question...Is there any way to use the app without making another copy of the database? My app is based on a static sqlite database that is never written to by the app. It's large... > 100mb. The app copies that into whatever location (Documents) so it ends up taking >200mg of space. Is there any way to not have it copy to some other location and just read directly from whereever it's stored in the bundle file? Thanks!

**Comment by andpor** [comment]
There maybe a solution for iOS...but would have to make it optional with a macro..

**Comment by andpor** [comment]
I have committed code changed to address both issues. by adding readOnly to the param list you will be able to point to app bundle db without requirign a copy. For locations, this is a backward incompatible change. default is now nosync location rather than docs. There are 32 choices for location param

1. default
2. Documents
3. Library

You will find fixes on master. Please test them out and let me know how you make out. These are iOS specific changes so Android should not be impacted.

**Comment by grittathh** [comment]
Thanks! The readOnly code works, except dealloc in SQLite.m needs some small changes I think (it crashes on reloading in the simulator with a "pointerValue" unrecognized error). This seems to fix it:

```
-(void)dealloc
{
  int i;
  NSArray *keys = [openDBs allKeys];
  NSDictionary *dbInfo;
  NSString *key;
  sqlite3 *db;

  /* close db the user forgot */
  for (i=0; i<[keys count]; i++) {
    key = [keys objectAtIndex:i];
    dbInfo = [openDBs objectForKey:key];
    db = [((NSValue *) dbInfo[@"dbPointer"]) pointerValue];
    sqlite3_close (db);
  }

#if !__has_feature(objc_arc)
  [openDBs release];
  [appDBPaths release];
  [super dealloc];
#endif
}
```

**Comment by andpor** [comment]
@grittathh - can you open a separate issue to track this change?

**Comment by andpor** [comment]
@scgough - can you test the location modifications?

**Comment by andpor** [comment]
resolved in 3.0.0
